src/App.js

```diff
@@ -13,12 +16,138 @@ const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const initialState = {
+  listIsVisible: true,
+  buttonIsVisible: false,
+  products: goodsFromServer,
+  selectedElementLength: 1,
+};
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = initialState
+
+  showList = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      listIsVisible: !prevState.listIsVisible,
+      buttonIsVisible: !prevState.buttonIsVisible,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      products: [...prevState.products].reverse(),
+    }));
+  }
+
+  sortAlphabet = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      products: [...prevState.products].sort(),
+    }));
+  }
+
+  reset = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      products: initialState.products,
```

Comment:
I think we should try use state instead of variables which are initialized outside

Reply:
Fixed!